lib/spack/llnl/util/link_tree.py

```diff
@@ -62,10 +63,16 @@ def find_conflict(self, dest_root, **kwargs):
                 return dest
         return None
 
-    def merge(self, dest_root, **kwargs):
+    def merge(self, dest_root, link=os.symlink, **kwargs):
         """Link all files in src into dest, creating directories
-           if necessary."""
+           if necessary.
+           If report_existing is True, do not break when the target exists but
+           rather return a list of files that could not be linked.
+           Note that files blocking directories will still cause an error.
+        """
         kwargs['order'] = 'pre'
+        report_existing = kwargs.get("report_existing", False)
```

Comment:
IMO this should also be called ```ignore_conflicts```. The variable definitely results in reporting existing files, but IMO the most important effect is that this option allows for conflicts to occur.